ReactNativeClient/lib/models/Tag.js

```diff
@@ -14,7 +14,13 @@ class Tag extends BaseItem {
 	}
 
 	static async noteIds(tagId) {
-		const rows = await this.db().selectAll('SELECT note_id FROM note_tags WHERE tag_id = ?', [tagId]);
+		// Get NoteIds of that are tagged with current tag or its descendants
+		const rows = await this.db().selectAll(`WITH RECURSIVE
+												parent_of(id, child_id) AS 
+												(SELECT id, id FROM tags where id=?
+												UNION ALL
+												SELECT parent_of.id, tags2.id FROM parent_of JOIN tags AS tags2 ON parent_of.child_id=tags2.parent_id)
+												SELECT note_id FROM note_tags WHERE tag_id IN (SELECT child_id from parent_of)`, [tagId]);
```

Comment:
Again I'd prefer if this is done in JavaScript rather than in SQL.

Reply:
Moved this to JS.